src/Core__Array.resi

```diff
@@ -29,55 +28,622 @@ let make: (~length: int, 'a) => array<'a>
 let fromInitializer: (~length: int, int => 'a) => array<'a>
 
 @val external isArray: 'a => bool = "Array.isArray"
-@get external length: array<'a> => int = "length"
+
+/**
+`length(array)` returns the `int` length (number of items) of the array.
+
+See [`Array.length`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length) on MDN.
+
+## Examples
+```rescript
+let someArray = ["hi", "hello"]
+
+Console.log(someArray->Array.length) // 2
+```
+*/
+@get
+external length: array<'a> => int = "length"
+
+// TODO: Docs
 @send external copyAllWithin: (array<'a>, ~target: int) => array<'a> = "copyWithin"
+
+// TODO: Docs
 @send
 external copyWithinToEnd: (array<'a>, ~target: int, ~start: int) => array<'a> = "copyWithin"
+
+// TODO: Docs
 @send
 external copyWithin: (array<'a>, ~target: int, ~start: int, ~end: int) => array<'a> = "copyWithin"
-@send external fillAllInPlace: (array<'a>, 'a) => unit = "fill"
-@send external fillInPlaceToEnd: (array<'a>, 'a, ~start: int) => unit = "fill"
-@send external fillInPlace: (array<'a>, 'a, ~start: int, ~end: int) => unit = "fill"
-@send external pop: array<'a> => option<'a> = "pop"
-@send external push: (array<'a>, 'a) => unit = "push"
-@variadic @send external pushMany: (array<'a>, array<'a>) => unit = "push"
-@send external reverseInPlace: array<'a> => unit = "reverse"
-@send external shift: array<'a> => option<'a> = "shift"
+
+/**
+`fillAllInPlace(array, value)` fills the entire provided `array` with the provided `value`.
+
+Beware this will *mutate* the array.
+
+See [`Array.fill`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill) on MDN.
+
+## Examples
+```rescript
+let myArray = [1, 2, 3, 4]
+myArray->Array.fillAllInPlace(9)
+
+Console.log(myArray) // [9, 9, 9, 9]
+```
+*/
+@send
+external fillAllInPlace: (array<'a>, 'a) => unit = "fill"
+
+/**
+`fillInPlaceToEnd` fills the entire provided `array` with the provided `value`, starting from the index provided to `start`.
```

Comment:
This is missing the argument specification. Also not sure that the filler words are all that necessary, in general.

```suggestion
`fillInPlaceToEnd(array, value, ~start)` fills `array` with `value` from the `start` index.
```